spec/controllers/sys_controller_spec.rb

```diff
@@ -28,208 +28,368 @@
+          @key = Setting.sys_api_key
+          @project = FactoryGirl.create(:project, :is_public => false)
+          @member = FactoryGirl.create(:member, :user => valid_user,
+                                                :roles => [commit_role],
+                                                :project => @project )
+          valid_user.save
+          request.env['HTTP_AUTHORIZATION'] = ActionController::HttpAuthentication::Basic.encode_credentials(valid_user.login, valid_user_password)
+        end
+
+        it "should respond 200 okay dokay for GET" do
+          post "repo_auth", { :key => @key, :repository => @project.identifier, :method => "GET", :git_smart_http => '1', :uri => "/git", :location => '/git' }
+          expect(response.code).to eq("200")
+        end
+
+        it "should respond 200 okay dokay for POST" do
+          post "repo_auth", { :key => @key, :repository => @project.identifier, :method => "POST", :git_smart_http => '1', :uri => "/git/#{@project.identifier}/git-receive-pack", :location => '/git' }
+          expect(response.code).to eq("200")
+        end
       end
-    end
 
-    describe '#repo_auth', 'for invalid credentials' do
-      before(:each) do
-        @key = Setting.sys_api_key
-        post 'repo_auth', key: @key, repository: 'any-repo', method: 'GET'
+      context 'for invalid login and user has role manager for project' do
+        before(:each) do
+          @key = Setting.sys_api_key
+          @project = FactoryGirl.create(:project, :is_public => false )
+          @member = FactoryGirl.create(:member, :user => valid_user,
+                                                :roles => [commit_role],
+                                                :project => @project)
+          request.env['HTTP_AUTHORIZATION'] = ActionController::HttpAuthentication::Basic.encode_credentials(valid_user.login, valid_user_password + "made invalid")
+          post "repo_auth", { :key => @key, :repository => @project.identifier, :method => "GET", :git_smart_http => '1', :uri => "/git", :location => '/git' }
+        end
+
+        it "should respond 401 auth required" do
+          expect(response.code).to eq("401")
+        end
       end
 
-      it 'should respond 401 auth required' do
-        expect(response.code).to eq('401')
-        expect(response.body).to eq('Authorization required')
+      context 'for valid login and user is not member for project' do
+        before(:each) do
+          @key = Setting.sys_api_key
+          @project = FactoryGirl.create(:project, :is_public => false)
+          request.env['HTTP_AUTHORIZATION'] = ActionController::HttpAuthentication::Basic.encode_credentials(valid_user.login, valid_user_password)
+          post "repo_auth", { :key => @key, :repository => @project.identifier, :method => "GET", :git_smart_http => '1', :uri => "/git", :location => '/git' }
+        end
+
+        it "should respond 403 not allowed" do
+          expect(response.code).to eq("403")
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.